src/main.rs

```diff
@@ -95,61 +84,15 @@ impl RunnerCache {
 
         match std::fs::write(cache_path, new_cache) {
             Ok(_) => (),
-            Err(e) => println!("Could not write toml to disk, error: {}", e),
+            Err(e) => eprintln!("Could not write toml to disk, error: {}", e),
         };
     }
 }
 
-pub fn main() {
-    let cli = Cli::parse();
-
-    let mut cache = RunnerCache::load();
-
-    let chosen;
-
-    // TODO: this is disgusting there must be a better way
-    if cli.force_choose_new {
-        chosen = select_new_runner();
-        if chosen.is_some() {
-            if cache.is_some() {
-                cache.as_mut().unwrap().add_runner(chosen.as_ref().unwrap());
-            }
-            else {
-                println!("Could not parse cache, intentionally not overwriting\
-                    , check it for errors.")
-            }
-        }
-    } else {
-        chosen = match cache {
-            Some(ref mut c) => match c.try_get_runner() {
-                Some(rnr) => Some(rnr), // runner found in the cache
-                None => { // runner not found in the cache
-                    let rnr = select_new_runner();
-                    if rnr.is_some() {
-                        c.add_runner(rnr.as_ref().unwrap());
-                    }
-                    rnr
-                },
-            },
-            None => select_new_runner(),
-        };
-    }
-
-
-    match chosen {
-        Some(cr) => cr.run(),
-        None => println!("No Runner Selected"),
-    };
-
-    println!("bye!");
-
-}
-
-/// Returns a Some(Runner) if the user selects one.
-///
-/// Uses `skim` to generate the TUI !this will change in future!
-fn select_new_runner() -> Option<Runner> {
-    let runners = runner::load_runners();
+fn select_new_runner(
```

Comment:
nit: please put this all on one line, its not long enough to be separated like this in my opinion